core/src/test/java/org/sonarsource/sonarlint/core/container/storage/StorageAnalyzerTest.java

```diff
@@ -19,71 +19,82 @@
  */
 package org.sonarsource.sonarlint.core.container.storage;
 
-import org.junit.Before;
-import org.junit.Rule;
-import org.junit.Test;
-import org.junit.rules.ExpectedException;
+import java.nio.file.Path;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
 import org.mockito.Mock;
 import org.mockito.MockitoAnnotations;
 import org.sonarsource.sonarlint.core.client.api.common.analysis.IssueListener;
 import org.sonarsource.sonarlint.core.client.api.connected.ConnectedAnalysisConfiguration;
 import org.sonarsource.sonarlint.core.client.api.connected.GlobalStorageStatus;
 import org.sonarsource.sonarlint.core.client.api.connected.ProjectStorageStatus;
 import org.sonarsource.sonarlint.core.client.api.exceptions.StorageException;
+import org.sonarsource.sonarlint.core.proto.Sonarlint;
 import org.sonarsource.sonarlint.core.util.ProgressWrapper;
 
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchThrowable;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 
-public class StorageAnalyzerTest {
-  @Rule
-  public ExpectedException exception = ExpectedException.none();
-
+class StorageAnalyzerTest {
   @Mock
   private GlobalUpdateStatusReader globalReader;
   @Mock
   private ProjectStorageStatusReader moduleReader;
   @Mock
   private ConnectedAnalysisConfiguration config;
+  @TempDir
+  Path tempDir;
 
+  private GlobalSettingsStore globalSettingsStore;
   private StorageAnalyzer analyzer;
 
-  @Before
+  @BeforeEach
   public void setUp() {
     MockitoAnnotations.initMocks(this);
     when(config.projectKey()).thenReturn("module1");
+    globalSettingsStore = new GlobalSettingsStore(new StorageFolder.Default(tempDir));
+    globalSettingsStore.store(Sonarlint.GlobalProperties.newBuilder().build());
     analyzer = new StorageAnalyzer(globalReader, moduleReader);
   }
 
   @Test
-  public void testNoGlobalStorage() {
-    when(globalReader.get()).thenReturn(null);
+  void testNoGlobalStorage() {
+    when(globalReader.read()).thenReturn(null);
+
+    Throwable exception = catchThrowable(() -> analyzer.analyze(mock(StorageContainer.class), config, mock(IssueListener.class), globalSettingsStore, new ProgressWrapper(null)));
```

Comment:
I replaced some of the assertThrows because I thought it was old-JUnit kind of assertions...

In general I prefer AssertJ assertions because they read more naturally, and you can assert more things on the exception. Now if you're talking specifically about `assertThrows` vs `catchThrowable` to execute a block and catch the error, I don't have a strong preference, but I would stay with AssertJ for consistency

Reply:
Ah, indeed, I missed that. Thanks for the clarification.